starter-code/src/components/AddProduct.js

```diff
@@ -0,0 +1,81 @@
+import React, {Component}  from 'react';
+
+export default class AddProduct extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      name: "",
+      calories: 0,
+      image: ""
+    };
+  }
+
+  // al hacer submit enviamos al padre las nuevas props y ponemos de nuevo a cero el state
+
+  handleFormSubmit = event => {
+    event.preventDefault();
+    this.props.addNewFood(this.state);
+    this.setState({ 
+      name: "",
+      calories: 0,
+      image: ""  
+     });
+  };
+
+  handleChange = (e) => (
+    this.setState({
+      [e.target.name]: e.target.value,
+      [e.target.calories]: e.target.value,
+      [e.target.image]: e.target.value
```

Comment:
No tiene mucho sentido que lo hagas de esa manera si utilizas `[e.target.name]`, `[e.target.calories]` y `[e.target.image]`. Si quieres coger el valor del atributo del mismo nombre, usa `[e.target.name]` y si quieres coger la clave del estado (como parece que quieres hacer) escribe directamente dicho nombre `this.setState({ name: e.target.value })`